### PR TITLE
Add ofInitialCapacity to primitive maps

### LINKED ISSUE #654: ofInitialCapacity for primitive map factories
As with MutableMapFactory, add this method for the primitive maps.

### PR DESCRIPTION
**Pull request by victornoel**
Closes #654

### CODE REVIEW

File: eclipse-collections-code-generator/src/main/resources/api/factory/map/mutableObjectPrimitiveMapFactory.stg [hunk 1]
```diff
@@ -38,6 +38,18 @@ public interface MutableObject<name>MapFactory
...
+     * Same as {@link #empty()}. but takes in an initial capacity
+     */
+
+    \<K> MutableObject<name>Map\<K> ofInitialCapacity(int capacity);
+
+    /**
+     * Same as {@link #empty()}. but takes in an initial capacity
+     */
+
+    \<K> MutableObject<name>Map\<K> withInitialCapacity(int capacity);
```
[the remark this answers is not in the document]
Comment (victornoel) [reply eclipse-collections-code-generator/src/main/resources/api/factory/map/mutableObjectPrimitiveMapFactory.stg]: ok, it's fixed now :)

[the remark this answers is not in the document]
Comment (victornoel) [reply eclipse-collections-code-generator/src/main/resources/api/factory/map/mutableObjectPrimitiveMapFactory.stg]: I've also fixed this in MutableMapFactory

**Comment by motlin** [comment]
LGTM :+1:

@nikhilnanivadekar might ask for the two commits to be squashed :)

**Comment by nikhilnanivadekar** [comment]
haha yes 😃 . Please squash the two commits.

**Comment by victornoel** [comment]
@nikhilnanivadekar done :)

**Comment by nikhilnanivadekar** [comment]
Thanks for your contribution!

**Pull request merged**